EmployeeList/Services/EmployeesAPIService.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  EmployeeListService.swift
+//  EmployeeList
+//
+//  Created by Wang Siman on 1/29/21.
+//  Copyright © 2021 None. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+final class EmployeesAPIService: APIServiceType {
+
+    internal let baseURL: String
+    internal let session: URLSession = URLSession.shared
+    internal let bgQueue: DispatchQueue = DispatchQueue.main
+    
+    init(baseURL: String =  "https://s3.amazonaws.com/sq-mobile-interview/") {
+        self.baseURL = baseURL
+    }
+ 
+    func call<Request>(from endpoint: Request) -> AnyPublisher<Request.ModelType, Error> where Request : APIRequestType {
+        do {
+            let request = try endpoint.buildRequest(baseURL: baseURL)
+            return session.dataTaskPublisher(for: request)
+                .retry(1)
+                .tryMap {
+                    guard let code = ($0.1 as? HTTPURLResponse)?.statusCode else {
```

Comment:
I would use labels, not numbers, to refer to these components. `$0.data`, etc

Reply:
This is also a good point! Appreciate it!